provision/docker/hc.go

```diff
@@ -63,6 +73,25 @@ func pingDockerRegistry(scheme string) error {
 	return nil
 }
 
+func newRequestWithCredentials(method, url string) (*http.Request, error) {
+	req, err := http.NewRequest(method, url, nil)
+	if err != nil {
+		return nil, err
+	}
+	var credentials string
+	if username, _ := config.GetString("docker:registry-auth:username"); username != "" {
+		credentials = username
+	}
+	if password, _ := config.GetString("docker:registry-auth:password"); password != "" {
+		credentials += ":" + password
+	}
+	if len(credentials) > 0 {
+		b64 := base64.StdEncoding.EncodeToString([]byte(credentials))
+		req.Header.Add("Authorization", "Basic "+b64)
```

Comment:
You can use `req.SetBasicAuth()` here instead of manually encoding the auth header.

Reply:
@cezarsa done